src/ansi-c/c_typecheck_type.cpp

```diff
@@ -239,16 +239,18 @@ void c_typecheck_baset::typecheck_type(typet &type)
             from_integer(4, size_type()));
       }
       else // give up, just use subtype
-        result=type.subtype();
+        result = to_type_with_subtype(type).subtype();
```

Comment:
I'm aware this outside the scope of this PR, but still: should this perhaps be marked `UNREACHABLE`?